lib/src/layer/polyline_layer/polyline_layer.dart

```diff
@@ -73,23 +73,22 @@ class PolylineLayer<R extends Object> extends StatefulWidget {
 }
 
 class _PolylineLayerState<R extends Object> extends State<PolylineLayer<R>> {
-  List<_ProjectedPolyline>? _cachedProjectedPolylines;
-  final _cachedSimplifiedPolylines = <int, List<_ProjectedPolyline>>{};
+  List<_ProjectedPolyline<R>>? _cachedProjectedPolylines;
+  final _cachedSimplifiedPolylines = <int, List<_ProjectedPolyline<R>>>{};
 
-  final _culledPolylines =
-      <_ProjectedPolyline>[]; // Avoids repetitive memory reallocation
+  double? _devicePixelRatio;
 
   @override
   void didUpdateWidget(PolylineLayer<R> oldWidget) {
     super.didUpdateWidget(oldWidget);
 
-    // Reuse cache
-    if (widget.simplificationTolerance != 0 &&
-        oldWidget.simplificationTolerance == widget.simplificationTolerance &&
-        listEquals(oldWidget.polylines, widget.polylines)) return;
-
-    _cachedSimplifiedPolylines.clear();
-    _cachedProjectedPolylines = null;
+    if (!listEquals(oldWidget.polylines, widget.polylines)) {
+      _cachedProjectedPolylines = null;
+      _cachedSimplifiedPolylines.clear();
+    } else if (!(widget.simplificationTolerance != 0 &&
+        oldWidget.simplificationTolerance == widget.simplificationTolerance)) {
+      _cachedSimplifiedPolylines.clear();
+    }
```

Comment:
Same here, quick description about the difference please.

Reply:
Fixed in a4fa7c9.